annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,223 @@
+                        }
+                        for (subtype in sortedSubtypes.keys) {
+                            tr {
+                                td {+"$subtype" }
+                                td {+"${annotationsPerSubtype[subtype]}"}
+                                if (previousAnnotationsPerSubtype[subtype] == null) {
+                                    td {+"n/a"}
+                                }
+                                else {
+                                    td {+"${annotationsPerSubtype[subtype]?.minus(previousAnnotationsPerSubtype[subtype]!!)}"}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            }
+        }
+        /**
+         * Search as far back as a week for a previous stats report
+         * The length of time can be changed
+         */
+        fun locatePreviousStats(currentDate: Calendar, statsDir: File): String {
+            val calendar = currentDate
+            val formatter = SimpleDateFormat("yyyy-MM-dd")
+            for (day in 1..7) {
+                calendar.add(Calendar.DAY_OF_YEAR, -1)
+                val lastDate = formatter.format(calendar.getTime())
+                val lastFilename = "$statsDir/StatsReport$lastDate.html"
+                if (File(lastFilename).exists()) {
+                    return lastFilename
+                }
+            }
+            return ""
+        }
+        fun getPreviousAnnotations(
+                statsFile: File,
+                userMap: MutableMap<String, Int>,
+                subtypeMap: MutableMap<String, Int>
+        ): Int {
+            val html: Document = Jsoup.parse(statsFile, "utf-8")
+            val firstTable: Element = html.selectFirst("table")
+            var rows: Elements = firstTable.select("tr")
+            var cols: Elements = Elements()
+            for (row in rows) {
+                cols = row.select("td")
+            }
+            val previousAnnotations: Int = cols.get(0).text().toInt()
+
+            // Annotations by user
+            val userTable: Element = html.select("table").get(1)
+            rows = userTable.select("tr")
+            for (row in rows) {
+                cols = row.select("td")
+                if (cols.isNotEmpty()) {
+                    userMap[cols[0].text()] = cols[1].text().toInt()
+                }
+            }
+
+            // Annotations by subtype
+            val subtypeTable: Element = html.select("table").get(2)
+            rows = subtypeTable.select("tr")
+            for (row in rows) {
+                cols = row.select("td")
+                if (cols.isNotEmpty()) {
+                    subtypeMap[cols[0].text()] = cols[1].text().toInt()
+                }
+            }
+            return previousAnnotations
+        }
+    }
+}
+
+data class Annotation(val documentName: String, val user: String, val subtype: String)
```

Comment:
💯 for pulling out a class for the annotated documents.  Since currently we are only tracking things at the document level, how about calling it `DocumentAnnotation`?

Reply:
Sounds good